src/dataframe/dataframe.jl

```diff
@@ -173,9 +173,11 @@ end
 index(df::DataFrame) = df.colindex
 columns(df::DataFrame) = df.columns
 
-# TODO: Remove these
-nrow(df::DataFrame) = ncol(df) > 0 ? length(df.columns[1])::Int : 0
-ncol(df::DataFrame) = length(index(df))
+Base.size(df::DataFrame) = ((length(df.columns) > 0 ? length(df.columns[1]) : 0),
+                            length(index(df)))
+Base.size(df::DataFrame, i::Int) = i == 1 ? 
```

Comment:
Would be cleaner with an `if... else` block (with a long-form function definition), the indentation a bit weird here. BTW, could print "$i" in the error message, that's always useful to understand what went wrong.